Shared/Samples/Add WMTS layer/AddWMTSLayerView.swift

```diff
@@ -0,0 +1,117 @@
+// Copyright 2025 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ArcGIS
+import SwiftUI
+
+@MainActor
+struct AddWMTSLayerView: View {
+    /// The view model for the sample.
+    @StateObject private var model = Model()
+    
+    /// The WMTS layer source selected by the picker.
+    @State private var selectedLayerSource = WMTSLayerSource.wmtsLayerInfo
+    
+    /// The error shown in the error alert.
+    @State private var error: Error?
+    
+    var body: some View {
+        MapView(map: model.map)
+            .task {
+                do {
+                    try await model.loadService()
+                    model.setWMTSLayer(for: selectedLayerSource)
+                } catch {
+                    self.error = error
+                }
+            }
+            .toolbar {
+                ToolbarItem(placement: .bottomBar) {
+                    Picker("WMTS Layer", selection: $selectedLayerSource) {
+                        ForEach(WMTSLayerSource.allCases, id: \.self) { source in
+                            Text(source.label)
+                        }
+                    }
+                    .onChange(of: selectedLayerSource) {
+                        model.setWMTSLayer(for: selectedLayerSource)
+                    }
+                }
+            }
+            .errorAlert(presentingError: $error)
+    }
+}
+
+private extension AddWMTSLayerView {
+    /// The view model that contains the map and WMTS layer.
+    @MainActor
```

Comment:
Up to you: Instead of marking the whole model main actor, consider mark the method main actor as there is only 1. Also, since there is only 2 to 3 instance properties, I'm OK with putting them in the view.